hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoTransferValidator.java

```diff
@@ -113,7 +113,8 @@ public void pureChecks(@NonNull final CryptoTransferTransactionBody op) throws P
             validateFalsePreCheck(nftIds.size() < nftTransfers.size(), TOKEN_ID_REPEATED_IN_TOKEN_LIST);
 
             // Verify that one and only one of the two types of transfers (fungible or non-fungible) is present
-            validateFalsePreCheck(!nonZeroFungibleValueFound && nftIds.isEmpty(), EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS);
+            validateFalsePreCheck(
+                    uniqueTokenAcctIds.isEmpty() && nftIds.isEmpty(), EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS);
             validateFalsePreCheck(nonZeroFungibleValueFound && !nftIds.isEmpty(), INVALID_ACCOUNT_AMOUNTS);
```

Comment:
Even here I think we should use `!uniqueTokenAcctIds.isEmpty()` @vtronkov  ?

Reply:
Can be done after this PR is merged

Reply:
@Neeharika-Sompalli when I change it `prechecksWork` test starts to fail.
Looking at the original code(`PureTransferSemanticChecks.checkTokenTransfersList`) we have the following logic:
```
final var ownershipChangesHere = scopedTransfers.getNftTransfersCount();
final var transferCounts = scopedTransfers.getTransfersCount();
if (ownershipChangesHere > 0) {
    ...
} else {
    if (transferCounts == 0) {
        return EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS;
    }
    count += transferCounts;
}
```
So we are returning `EMPTY_TOKEN_TRANSFER_ACCOUNT_AMOUNTS` only when _ownershipChangesHere_ and _transferCounts_ are 0. This seems to correlate to both `uniqueTokenAcctIds` and `nftIds` lists being empty in the new code base. What do you think?

Reply:
Thanks @vtronkov . 
`transferCounts` is just seeing if the transferList has any items. But `uniqueTokenAcctIds` is checking if they are valid.  I will look at that change and `prechecksWork`